src/main/java/de/tum/in/www1/exerciseapp/domain/DragAndDropSubmittedAnswer.java

```diff
@@ -70,6 +70,47 @@ public DragItem getSelectedDragItemForDropLocation(DropLocation dropLocation) {
         return null;
     }
 
+    /**
+     * Check if a dragItem or dropLocation were deleted and delete reference to in mappings
+     * @param question the changed question with the changed DragItems and DropLocations
+     */
+    private void checkForDeletedMappings(DragAndDropQuestion question) {
+
+        if( question != null) {
+            // Check if a dragItem or dropLocation was deleted and delete reference to it in mappings
+            Set<DragAndDropMapping> selectedMappingsToDelete = new HashSet<>();
+            for (DragAndDropMapping mapping : this.getMappings()) {
+                if ((!question.getDragItems().contains(mapping.getDragItem())) ||
+                    (!question.getDropLocations().contains(mapping.getDropLocation()))) {
+                    selectedMappingsToDelete.add(mapping);
+                }
+            }
+            for (DragAndDropMapping mappingToDelete: selectedMappingsToDelete) {
+                this.removeMappings(mappingToDelete);
+            }
+        }
+    }
+
+    /**
+     * Delete all references to question, dragItems and dropLocations if the question was changed
+     *
+     * @param quizExercise the changed quizExercise-object
+     */
+    public void updateForDeletedReferences (QuizExercise quizExercise) {
```

Comment:
this method name "updateForDeletedReferences" is confusing, because something is actually deleted (according to the JavaDoc comment). Please rename the method name